### ISSUE TITLE
YouTube embed in Independent Publisher 2 is obscuring the block below it

### ISSUE THREAD

**Issue by p-jackson** [Automattic/wp-calypso#55357]
### Steps to reproduce the behavior

1. Create a new post
2. Paste this content into the code editor
```html
<!-- wp:embed {"url":"https://www.youtube.com/watch?v=QH2-TGUlwu4","type":"rich","providerNameSlug":"youtube","responsive":true,"className":"wp-embed-aspect-16-9 wp-has-aspect-ratio"} -->
<figure class="wp-block-embed is-type-rich is-provider-youtube wp-block-embed-youtube wp-embed-aspect-16-9 wp-has-aspect-ratio"><div class="wp-block-embed__wrapper">
https://www.youtube.com/watch?v=QH2-TGUlwu4
</div></figure>
<!-- /wp:embed -->

<!-- wp:list -->
<ul><li>this post</li><li>has a</li><li>bunch of</li><li>bullet points</li><li>one</li><li>two</li><li>three</li><li>four</li></ul>
<!-- /wp:list -->
```
3. Publish
4. Switch theme to Independent Publisher 2
5. View on frontend

#### What I expected to happen
All the bullet points should appear below the YouTube video. This is what it looks like in the TT1 theme:
<img width="685" alt="Screenshot 2021-08-11 at 5 52 01 PM" src="https://user-images.githubusercontent.com/1500769/128976571-e6c64456-6d4f-419a-8cf4-57b85fa7578a.png">

#### What actually happened
Some of the bullet points are obscured because the YouTube video is overtop of it.

<img width="782" alt="Screenshot 2021-08-11 at 5 52 53 PM" src="https://user-images.githubusercontent.com/1500769/128976655-36690d75-0676-40b7-a512-c9a49b06481e.png">

#### Browser / OS version
Reproduced on Mac with Edge, Firefox and Safari

#### Is this specific to the applied theme? Which one?
Independent Publisher 2

#### Does this happen on simple or atomic sites or both?
Not happening on atomic 🤔 

#### Is there any console output or error text?
Nothing relevant

#### Level of impact (Does it block purchases? Does it affect more than just one site?)
Discovered when testing typical blogger flows. Independent Publisher 2 is a popular blogging theme, and blocking post content is high impact IMO.

#### Reproducibility (Consistent, Intermittent) Leave empty for consistent.
Every time (on simple sites)

**Comment by arthur791004** [comment]
After investigating, the problem is because of `is-type-rich`. 

Our blocks have following styles
```css
.wp-block-embed.is-type-video iframe,
.wp-block-embed.is-type-video iframe {
	max-height: 100%;
}
```
See: https://github.com/Automattic/themes/blob/b22e12b419835a8f286a30228a22a703386df738/photos/blocks.css#L42

However, we use `translateOEmbedApiRequest` to translated oEmbed requests to standard embed requests on the **SIMPLE SITES**. But it doesn't return any type and fallback to default 'rich' so that the embed video style is broken. Besides, there is another problem that both of our embed Vimeo and embed Dailymotion don't work correctly because we don't register any handler for these two.

Here are possible solutions:

1. Remove `translateOEmbedApiRequest` and `wpcom_wrap_embeds_in_a_div` because it seems to work well in **AT SITES** without those two. But there are some histories about this one and maybe there are some edge cases I don't know. Here are related issues:
   * https://github.com/Automattic/wp-calypso/issues/43047
   * https://github.com/Automattic/wp-calypso/issues/9605

2. I'm not sure but maybe we could merge the response from both two to get the correct type and reduce some risk.

3. Add the same style to `is-type-rich`. But I think it might not make sense because the rich type might contain other elements and it's not suitable to add the same style to it.


I found that @jeherve @ockham might know more about `translateOEmbedApiRequest`. Do you think is it possible to remove it? Or is there any better solution for this one? Thank you!

**Comment by roo2** [comment]
I put some logging in to confirm what's going on

It looks like for the _independent publisher 2_ theme, `current_theme_supports( 'jetpack-responsive-videos' )` is returning true which causes `jetpack_responsive_videos_embed_html` to be called which ultimately wraps the video in a `'<div class="jetpack-video-wrapper">'` block. This doesn't happen for the working "_blank canvas_" theme.

I saw that`translateOEmbedApiRequest` is loaded in the block editor. I loaded the block editor with the broken _Indipendent publisher 2_ theme and the working _blank canvas_ theme. In both cases, the `translateOEmbedApiRequest` resolved the same html. In my case `"<iframe class="youtube-player" width="640" height="360" src="https://www.youtube.com/embed/QH2-TGUlwu4?version=3&amp;rel=1&amp;showsearch=0&amp;showinfo=1&amp;iv_load_policy=1&amp;fs=1&amp;hl=en&amp;autohide=2&amp;wmode=transparent" allowfullscreen="true" style="border:0;" sandbox="allow-scripts allow-same-origin allow-popups allow-presentation"></iframe>"` so I don't think that this function is where the problem is. 

I'm currently looking into responsive-videos.js which is giving the iframe its height: https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/responsive-videos/responsive-videos.js and confirming that this only affects wpcom sites


> Besides, there is another problem that both of our embed Vimeo and embed Dailymotion don't work correctly because we don't register any handler for these two.

Do you mean it's completely broken, or is it just that overlapping content like this youtube issue? In which case we can probably fix it as part of this, whatever we decide on

**Comment by arthur791004** [comment]
@roo2 

> It looks like for the independent publisher 2 theme, current_theme_supports( 'jetpack-responsive-videos' ) is returning true which causes jetpack_responsive_videos_embed_html to be called which ultimately wraps the video in a '<div class="jetpack-video-wrapper">' block. This doesn't happen for the working "blank canvas" theme.

Great point! The `jetpack-responsive-video` will get the width from parent and apply to iframe. But the independent publisher 2 sets the width to `jetpack-video-wrapper` to `1100px` at [here](https://github.com/Automattic/independent-publisher-2-wc/blob/ba17aa477e5f7b51613daa39ff42fc2ae27fbccc/style.css#L3270) and causes the height of iframe overflow the content because the height depends on the width of `wp-block-embed__wrapper`. However, `jetpack-responsive-videos` still works well on the AT site, so IMHO we should

1. Fix this problem without changing any styles for the independent publisher 2 theme.
2. Adjust styles for the independent publisher 2 theme to make video larger than original one because it's the feature of that theme: **Featured Images and other images greater than 1100px wide will display in a big, bold way, outdented from the surrounding content.**

Does it make sense?

> Do you mean it's completely broken, or is it just that overlapping content like this youtube issue? In which case we can probably fix it as part of this, whatever we decide on

I tested Vimeo and Dailymotion on the simple site before and got the response `invalid_embed` from API. Thus, it becomes link rather than embed preview. But It works now 😅

Yeah, but they also overlap the content like this youtube issue.

**Comment by roo2** [comment]
@arthur791004 

>But the independent publisher 2 sets the width to jetpack-video-wrapper to 1100px at here

Ahh so that's what's happening!

I think we need to change the way that independent publisher increases the size of the video to a way that is compatible with the `jetpack-responsive-videos` script. If we could get the `margin-left: -180px; width: 1100px` rule applied to the parent element (`.wp-block-embed__wrapper`) then I think that would fix it.

I've created a test in my sandbox to change 
`	body:not(.has-sidebar) .jetpack-video-wrapper,`
to:
`	body:not(.has-sidebar) .wp-block-embed__wrapper,`

in [indpendent-publisher/style.css:3264](https://github.com/Automattic/independent-publisher-2-wc/blob/ba17aa477e5f7b51613daa39ff42fc2ae27fbccc/style.css#L3264)
This fixes the issue but I just need to check that this would make sense for embeded content with the `wp-block-embed__wrapper` class other than videos

**Comment by jeherve** [comment]
> I think we need to change the way that independent publisher increases the size of the video to a way that is compatible with the jetpack-responsive-videos script.

Would it be an option to remove support for Jetpack Responsive Videos in the theme, and instead switch to the more recent and more block-compatible core feature, Responsive embeds?
https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/#responsive-embedded-content

**Comment by roo2** [comment]
@jeherve

it looks like the theme has enabled responsive embeds  via `add_theme_support( 'responsive-embeds' );` https://github.com/Automattic/themes/blob/trunk/independent-publisher-2/functions.php#L101

But I see it also supports `jetpack-responsive-videos` https://github.com/Automattic/themes/blob/trunk/independent-publisher-2/inc/jetpack.php#L52

I'm not sure what the difference is between `jetpack-responsive-video`  and `responsive-embeds` and what effect removing  `jetpack-responsive-video` support would have on the theme. 

We're trying to maintain this theme feature where large images or videos become "outdented" so that they stand out from the other content.

e.g. 

<img width="1157" alt="Screen Shot 2021-08-25 at 9 31 46 pm" src="https://user-images.githubusercontent.com/22446385/130783345-25df2cf6-a3b8-4278-b0d0-61032272d432.png">

**Comment by jeherve** [comment]
> it looks like the theme has enabled responsive embeds via `add_theme_support( 'responsive-embeds' );` [https://github.com/Automattic/themes/blob/trunk/independent-publisher-2/functions.php#L101](https://github.com/Automattic/themes/blob/trunk/independent-publisher-2/functions.php?rgh-link-date=2021-08-25T11%3A27%3A05Z#L101)
> 
> But I see it also supports `jetpack-responsive-videos` [https://github.com/Automattic/themes/blob/trunk/independent-publisher-2/inc/jetpack.php#L52](https://github.com/Automattic/themes/blob/trunk/independent-publisher-2/inc/jetpack.php?rgh-link-date=2021-08-25T11%3A27%3A05Z#L52)

In Jetpack's Responsive videos we try to account for the Core Responsive embeds and bail:
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/modules/theme-tools/responsive-videos.php#L27-L30

Maybe we should pick one and stick it to it in the theme.

> I'm not sure what the difference is between jetpack-responsive-video and responsive-embeds

`jetpack-responsive-videos` is an old implementation, pre-Gutenberg, and was meant to ensure that video embeds of different sources would all stick to the content area while keeping the video's aspect ratio. See p1HpG7-1rq-p2 for more background on it.

Responsive Embeds is a feature that was added to Core recently, and that is built for embeds created via the block editor.

> what effect removing jetpack-responsive-video support would have on the theme.

Removing Jetpack Responsive Videos support would have an impact on videos that were added in posts using the Classic editor (i.e. old, pre-Gutenberg posts, posts using the Classic block, and posts created on sites using the Classic editor plugin).

> We're trying to maintain this theme feature where large images or videos become "outdented" so that they stand out from the other content.

Since those large embeds can only be created via a block and its width settings, I think Responsive Embeds is the tool for the job here.

It's probably a larger discussion to have, but maybe we should slowly start deprecating Jetpack Responsive Videos in our themes and instead support Core's solution instead.

**Comment by roo2** [comment]
@jeherve I put logging in the `jetpack-responsive-videos` code that is responsible for removing the plugin if `responsive embeds is enabled  (https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/modules/theme-tools/responsive-videos.php#L156

In the case of this youtube video it's looking for a `'core-embed'` block, but I see it's being called with a `'core/embed'` block (`/` instead of `-`) so not deactivating the plugin for that block. That looks like it might be a bug/oversight to me, what do you think?

**Comment by jeherve** [comment]
> it's looking for a 'core-embed' block, but I see it's being called with a 'core/embed' block (/ instead of -) so not deactivating the plugin for that block

Oh, nice catch! I created a PR to fix this here:
https://github.com/Automattic/jetpack/pull/20834

**Comment by roo2** [comment]
Nice @jeherve that fixes this issue! 🚀

Closed by Automattic/jetpack#20834
